configs/defs/rocky.textproto

```diff
@@ -1384,3 +1384,4017 @@ benchmark_configs: {
+      "                        expected_regex: \"action_mail_acct\\\\s*=\\\\s*root\""
+      "                      }"
+      "                    }"
+      "            }"
+      "            file_checks:{"
+      "                files_to_check:{single_file:{path:\"/etc/audit/auditd.conf\"}}"
+      "                    content_entry:{"
+      "                      match_type: ALL_MATCH_ANY_ORDER"
+      "                      match_criteria: {"
+      "                        filter_regex: \"\\\\s*admin_space_left_action(\\\\s|=).*\""
+      "                        expected_regex: \"admin_space_left_action\\\\s*=\\\\s*(halt|single)\""
+      "                      }"
+      "                    }"
+      "            }"
+      "        }"
+      "    }"
+  }
+}
+# The image check looks at the .rules files in /etc/audit/rules.d.
+# The instance check looks at the rule file (/etc/audit/audit.rules) which is automatically compiled from the files in /etc/audit/rules.d.
+# It is not possible to read the rules while auditd is running since they are provided directly to the kernel.
+benchmark_configs: {
+  id: "changes-to-sudoers-collected"
+  compliance_note: {
+    version: { cpe_uri: "cpe:/o:rockylinux:rockylinux:8.5" version: "1.0.0" benchmark_document: "CIS Rocky Linux 8" }
+    title: "Ensure changes to system administration scope (sudoers) is collected"
+    description: 
+      "Monitor scope changes for system administrators. If the system has been properly "
+      "configured to force system administrators to log in as themselves first and then use the "
+      "sudo command to execute privileged commands, it is possible to monitor changes in scope. "
+      "The file /etc/sudoers, or files in /etc/sudoers.d, will be written to when the file(s) or "
+      "related attributes have changed. The audit records will be tagged with the identifier "
+      "\"scope\"."
+    rationale:
+      "Changes in the /etc/sudoers and /etc/sudoers.d files can indicate that an unauthorized "
+      "change has been made to the scope of system administrator activity"
+    remediation:
+      "Edit or create a file in the /etc/audit/rules.d/ directory, ending in .rules extension, with "
+      "the relevant rules to monitor scope changes for system administrators.\n"
+      "Example:\n"
+      "```\n"
+      "# printf \"\n"
+      "-w /etc/sudoers -p wa -k scope\n"
+      "-w /etc/sudoers.d -p wa -k scope\n"
+      "\" >> /etc/audit/rules.d/50-scope.rules\n"
+      "```\n"
+      "Merge and load the rules into active configuration:\n"
+      "```\n"
+      "# augenrules --load\n"
+      "```\n"
+      "Check if reboot is required.\n"
+      "```\n"
+      "# if [[ $(auditctl -s | grep \"enabled\") =~ \"2\" ]]; then printf \"Reboot "
+      "required to load rules\n\"; fi\n"
+      "```"
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "scan_type_specific:{"
+      "    instance_scanning:{"
```

Comment:
A comment above the check outlines the reasoning:

```
# The image check looks at the .rules files in /etc/audit/rules.d.
# The instance check looks at the rule file (/etc/audit/audit.rules) which is automatically compiled from the files in /etc/audit/rules.d.
# It is not possible to read the rules while auditd is running since they are provided directly to the kernel.
```

The "hierarchy" we found is that the rule files in `/etc/audit/rules.d` are compiled in the single `/etc/audit/audit.rules`, which is then loaded by auditd.
In a running instance, we believe the `audit.rules` is closer to rules loaded, but in a base image the `audit.rules` may not be present or may be overwritten (recompiled from the `rules.d` subfolder), thus we assumed a safer approach to inspect those files.

Reply:
I see, that makes sense. Thanks for the clarification! Should be good as is then.